mqtt/mqtt-edgehub/src/edgelet/workload/client.rs

```diff
@@ -0,0 +1,136 @@
+use bytes::buf::BufExt;
+use chrono::{DateTime, Utc};
+use hyper::{body, Body, Client};
+
+use crate::edgelet::{
+    make_hyper_uri, ApiError, CertificateResponse, Connector, Scheme, ServerCertificateRequest,
+};
+use http::{Request, StatusCode};
+
+pub struct WorkloadClient {
+    client: Client<Connector, Body>,
+    scheme: Scheme,
+}
+
+impl WorkloadClient {
+    pub(crate) fn new(client: Client<Connector, Body>, scheme: Scheme) -> Self {
+        Self { client, scheme }
+    }
+
+    pub async fn create_server_cert(
+        &self,
+        module_id: &str,
+        generation_id: &str,
+        hostname: &str,
+        expiration: DateTime<Utc>,
+    ) -> Result<CertificateResponse, WorkloadError> {
+        let path = format!(
+            "/modules/{}/genid/{}/certificate/server?api-version=2019-01-30",
+            module_id, generation_id
+        );
+        let uri =
+            make_hyper_uri(&self.scheme, &path).map_err(|e| ApiError::ConstructRequestUrl(e))?;
+
+        let req = ServerCertificateRequest::new(hostname.to_string(), expiration.to_rfc3339());
+        let body = serde_json::to_string(&req).map_err(ApiError::SerializeRequestBody)?;
+        let req = Request::post(uri)
+            .body(Body::from(body))
+            .map_err(ApiError::ConstructRequest)?;
+
+        let res = self
+            .client
+            .request(req)
+            .await
+            .map_err(ApiError::ExecuteRequest)?;
+
+        if res.status() != StatusCode::OK {
+            return Err(ApiError::UnsuccessfulResponse(res.status()).into());
```

Comment:
nit: can we get more than just a status? does workload api provides error description in the response body?

Reply:
I think a can compose a string and put in the error value